src/workerd/api/node/tests/process-exit-test.js

```diff
@@ -0,0 +1,26 @@
+import { rejects, ok } from 'assert';
+
+let called = false;
+
+export const test = {
+  async test(_, env) {
+    // We don't really have a way to verifying that the correct log messages
+    // we emitted. For now, we need to manually check the log out, and here
+    // we check only that we received an internal error and that no other
+    // lines of code ran after the process.exit call.
+    await rejects(env.subrequest.fetch('http://example.org'), {
+      message: /^The Node.js process.exit/,
+    });
+    ok(!called);
+  },
+};
+
+export default {
+  async fetch() {
+    try {
+      process.exit(123);
```

Comment:
I think `process.exit(1)` and `process.exit(0)` should make a difference.

Reply:
A difference in what way?

Reply:
Maybe we should behave differently depending on the exit codes, such as if 0, we succeed and keep the behavior as is, and for 1 (failure) do something else?

Reply:
Yeah, but behave differently *how*?  In either case the call to `process.exit(...)` is expected to be terminal in a way we do not support but need to approximate. There's not expected to be any way of handling the exit outside of `process.on('exit')` which doesn't really change anything about the behavior. Either way we need to stop JS from executing further on this request and need to shut down the IoContext. Really not sure what else we can do here so open to more specific suggestions ;-)

Reply:
I don't have any suggestions on this, and once I thought about this, I realized that the behavior shouldn't change.